src/Container.php

```diff
@@ -110,53 +109,46 @@ private function isUserDefined(string $id): bool
     }
 
     /**
-     * @param class-string $qn
+     * @param class-string $class
      *
      * @return object
      * @throws ContainerException
      */
-    private function instantiate(string $qn): object
+    private function instantiate(string $class): object
     {
-        if (isset($this->beingResolved[$qn])) {
-            $msg = sprintf('Cyclic dependency of %s.', $qn);
-            throw new ContainerException($this->buildExceptionReport($msg));
+        if (isset($this->beingResolved[$class])) {
+            throw new ContainerException("Cyclic dependency of [$class].");
         }
 
-        $this->beingResolved[$qn] = true;
+        $this->beingResolved[$class] = true;
 
-        $reflection = new ReflectionClass($qn);
+        $reflection = new ReflectionClass($class);
 
         if (!$reflection->isInstantiable()) {
-            $reason = sprintf('%s is not instantiable.', $qn);
-            throw new ContainerException($this->buildExceptionReport($reason));
+            throw new ContainerException("[$class] is not instantiable.");
         }
 
         $constructor = $reflection->getConstructor();
         // No constructor defined or constructor requires no arguments? Just create new instance.
         if (!$constructor || !$constructor->getParameters()) {
-            return new $qn();
+            return new $class();
         }
 
-        try {
-            $instance = new $qn(...$this->resolveParams($qn, $constructor));
-            unset($this->beingResolved[$qn]);
-            return $instance;
-        } catch (Throwable $exception) {
-            $message = sprintf('Failed to instantiate %s.', $qn);
-            throw new ContainerException($message, 0, $exception);
```

Comment:
We don't need this. We are basically catching `ContainerException` and wrapping it into another `ContainerException`.